cloudflare-tag-purge.php

```diff
@@ -9,251 +9,271 @@
+			// on production: no prefixes are added for the tags
+			return null;
+		}
+
+		$prefix = str_replace( '.yoast.com', '', $this->read_environment_setting( 'DOMAIN_CURRENT_SITE', 'staging-local.yoast.com' ) );
+		$prefix = str_replace( '.', '-', $prefix );
+
+		return trim( strtolower( $prefix ) ) . '-';
+	}
+
+
+	/**
+	 * @param string $level
+	 * @param array  $data
+	 */
+	private function cache_tag_log( string $level, array $data ) {
+		$log_location = $this->read_environment_setting( 'CF_LOG_PATH', null );
+		if ( $log_location === null ) {
+			error_log( 'Cloudflare log file path not configured (CF_LOG_PATH)' );
+
+			return;
+		}
+
+		$data['level']   = $level;
+		$data['server']  = $_SERVER['SERVER_NAME'];
+		$data['site']    = $this->read_environment_setting( 'DOMAIN_CURRENT_SITE', 'staging-local.yoast.com' );
+		$data['created'] = \date( 'Y-m-d H:i:s' );
+
+		$file_stream = @fopen( $log_location, 'a' );
+		@fwrite( $file_stream, \wp_json_encode( $data ) . "\n" );
+		@fclose( $file_stream );
+
+		if ( ! is_writable( $log_location ) ) {
+			error_log( 'Cloudflare log file path not writable: ' . $log_location );
+
+			return;
+		}
+	}
+
+	/**
+	 * @param $classes
+	 *
+	 * @return mixed
+	 */
+	private function cache_prefix_body_class( $classes ) {
+		if ( $this->get_cache_prefix() === null ) {
+			// return default classes, without prefix (production)
+			return $classes;
+		}
+
+		// return default classes + prefixed classes
+		$prefixedClasses = $classes;
+		foreach ( $prefixedClasses as $key => $class ) {
+			$prefixedClasses[ $key ] = $this->get_cache_prefix() . $class;
+		}
+
+		return array_merge( $classes, $prefixedClasses );
+	}
+
+	/**
+	 * Get the sitemap url, and flush by url
+	 *
+	 * @param int $post_id
+	 *
+	 * @return string
+	 */
+	private function get_sitemap_url( int $post_id ) {
+		$prefix    = get_site_url() . '/';
+		$post_type = get_post_type( $post_id );
+
+		if ( empty( $post_type ) ) {
+			return $prefix . 'sitemap_index.xml';
+		}
+
+		return $prefix . $post_type . '-sitemap.xml';
+	}
+
+	/**
+	 * @param int     $post_id
+	 * @param WP_Post $post
+	 */
+	private function execute_tag_purge( int $post_id, $post ) {
+		$tags = $this->get_cache_tags_by_post_id( $post_id );
+
+		if ( count( $tags ) === 0 ) {
+			$this->cache_tag_log( YOAST_CLOUDFLARE_LOG_STATUS_INFO, [
+				'purge'   => false,
+				'info'    => 'No tags found for post #' . $post_id,
+				'post_id' => $post_id
+			] );
+
+			return;
+		}
+
+		$loops = ceil( count( $tags ) / 30 ); // a maximum of 30 tags per API call
+		for ( $i = 0; $i <= $loops; $i ++ ) {
+			$api_tags = array_splice( $tags, ( $i * 30 ), 30 );
+
+			if ( \count( $api_tags ) === 0 ) {
+				continue;
+			}
+
+			$this->cloudflare_cache_clear( [ 'tags' => array_values( $api_tags ) ] );
+		}
+
+		$sitemap_url = $this->get_sitemap_url( (int) $post_id );
+
+		if ( ! empty( $sitemap_url ) ) {
+			$this->cloudflare_cache_clear( [ 'files' => [ $sitemap_url ] ] );
```

Comment:
I think we should always clear the sitemap index cache and also the sitemap of the specific object_types